boa_wasm/src/lib.rs

```diff
@@ -1,22 +1,11 @@
-use boa::{exec::Executable, parse, Context};
+use boa::Context;
 use wasm_bindgen::prelude::*;
 
 #[wasm_bindgen]
 pub fn evaluate(src: &str) -> Result<String, JsValue> {
     // Setup executor
-    let mut context = Context::new();
-
-    let expr = match parse(src, false) {
-        Ok(res) => res,
-        Err(e) => {
-            return Err(format!(
-                "Uncaught {}",
-                context.construct_syntax_error(e.to_string()).display()
-            )
-            .into());
-        }
-    };
-    expr.run(&mut context)
+    Context::new()
+        .eval(src)
```

Comment:
As we talked in Discord, this should still call the parser and the execution separately, now parser, compilation and execution. This allows better control for the Test262 suite, for example, and for benchmarking.